internal/app/upstream/multiplexer.go

```diff
@@ -0,0 +1,34 @@
+package multiplexer
+
+import (
+	"context"
+
+	v2 "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	"google.golang.org/grpc"
+)
+
+// Multiplexer handles the connections to the upstream control plane
+type Multiplexer interface {
+	// QueueRequest starts a stream with the upstream control plane
+	QueueRequest(context.Context, chan *v2.DiscoveryRequest, chan *v2.DiscoveryResponse)
+}
+
+type multiplexer struct {
+	conn *grpc.ClientConn
+}
+
+// NewMux creates an instance based on the typeUrl of the resource
+// A new instance of Multiplexer is recommended per xds type
+// Eg. For eds requests of different services, create an instance each.
```

Comment:
grammar nits: 
- `s/Eg./e.g.,/`
- lowercase `For`
- capitalize `eds`

How about invert the sentence order? More specifically, I'm suggesting we rephrase this third line as:
```suggestion
// e.g., create an instance for each EDS request of different services
```